Source/Relativity.Test.Helpers/Relativity.Test.Helpers/Kepler/OAuth2Helper.cs

```diff
@@ -0,0 +1,164 @@
+
+			dynamic data = JObject.Parse(tokenResponse);
+			string bearerToken = data.access_token.Value;
+			return bearerToken;
+		}
+
+		/// <summary>
+		/// Use this to cleanup OAuth2 during tests
+		/// </summary>
+		/// <param name="clientId"></param>
+		/// <returns></returns>
+		public async Task DeleteOAuth2ClientAsync(string clientId)
+		{
+			await _oAuth2ClientManager.DeleteAsync(clientId);
+		}
+
+		/// <summary>
+		/// Retrieves User info by email
+		/// </summary>
+		/// <param name="email"></param>
+		/// <returns></returns>
+		private DTOs.User GetUser(string email)
+		{
+			_rsapiClient.APIOptions = new APIOptions { WorkspaceID = -1 };
+
+			DTOs.User user;
+
+			Condition userQueryCondition = new TextCondition(DTOs.UserFieldNames.EmailAddress, TextConditionEnum.EqualTo, email);
+
+			DTOs.Query<DTOs.User> userQuery = new DTOs.Query<DTOs.User>(DTOs.FieldValue.AllFields, userQueryCondition, new List<Sort>());
+
+			DTOs.QueryResultSet<DTOs.User> resultSet = _rsapiClient.Repositories.User.Query(userQuery);
+
+			if (resultSet.Success && resultSet.TotalCount > 0)
+			{
+				user = resultSet.Results.First().Artifact;
+			}
+			else
+			{
+				throw new Exception($"Could not find user {email}");
+			}
+
+			return user;
+		}
+
+		/// <summary>
+		/// Simply gets a bearer token from Relativity
+		/// </summary>
+		/// <param name="tokenUrl"></param>
+		/// <param name="clientId"></param>
+		/// <param name="clientSecret"></param>
+		/// <param name="scope"></param>
+		/// <param name="grantType"></param>
+		/// <returns></returns>
+		private async Task<string> GetTokenAsync(string tokenUrl, string clientId, string clientSecret, string scope, string grantType)
+		{
+			HttpClient client = new HttpClient();
+			var httpContent = new HttpRequestMessage(HttpMethod.Post, tokenUrl);
+
+			client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/x-www-form-urlencoded"));
+
+			var values = new List<KeyValuePair<string, string>>();
+			values.Add(new KeyValuePair<string, string>("client_id", clientId));
+			values.Add(new KeyValuePair<string, string>("client_secret", clientSecret));
+			values.Add(new KeyValuePair<string, string>("scope", scope));
+			values.Add(new KeyValuePair<string, string>("grant_type", grantType));
+
+			httpContent.Content = new FormUrlEncodedContent(values);
+
+			HttpResponseMessage response = await client.SendAsync(httpContent);
+
+			if (response.IsSuccessStatusCode)
+			{
+				return await response.Content.ReadAsStringAsync();
+			}
+
+			Console.WriteLine("Token Generation failed.");
+
+			return null;
+		}
+
+		/// <summary>
+		/// Returns the URL for Token generation for oAuth
+		/// </summary>
+		/// <returns></returns>
+		private string GetTokenUrl(string protocol, string serverAddress)
+		{
+			return $"{protocol}://{serverAddress.ToLower().Replace("-services", "")}/Relativity/Identity/connect/token";

```

Comment:
can we add a comment here for why we are removing -services. This will help the developer who looks at this code in future.